system-x/services/ldap/src/main/java/software/tnb/ldap/service/Ldap.java

```diff
@@ -0,0 +1,63 @@
+package software.tnb.ldap.service;
+
+import software.tnb.common.account.NoAccount;
+import software.tnb.common.client.NoClient;
+import software.tnb.common.service.Service;
+import software.tnb.ldap.validation.LdapValidation;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+public class Ldap extends Service<NoAccount, NoClient, LdapValidation> {
```

Comment:
this should be an abstract class and there should be a separate `LocalLdap` class that takes care of running the container

for example like [this](https://github.com/tnb-software/TNB/blob/main/system-x/services/jms/amq/src/main/java/software/tnb/jms/amq/service/AMQBroker.java#L9) and [this](https://github.com/tnb-software/TNB/blob/main/system-x/services/jms/amq/src/main/java/software/tnb/jms/amq/resource/local/LocalAMQBroker.java#L19-L20)

Reply:
Done